scalyr_agent/util.py

```diff
@@ -677,6 +679,46 @@ def get_web_url_from_upload_url(server):
     return server
 
 
+def parse_data_rate_string(value):
+    """Return a rate in bytes per second parsed from a string of a float or int followed by a unit.
+    Takes into account bit (`b`) vs byte (`B`), ignores capitalization of things like `k` vs `K` and the time unit
+    denominator. Allowed numerators go up to terabytes, and allowed denominators go up to weeks.
+
+    :param value: String with value and unit
+    :return: The parsed rate in bytes per second, or raise ValueError if it could not be parsed
+    """
+    m = re.search(r"(-?\d+\.?\d*)\s*([kKmMgGtT]?)([bB])/([sSmMhHdDwW])", value)
+    if m:
+        value = float(m.group(1))
+        numerator = m.group(2).upper()
+        bit_or_byte = m.group(3)
+        denominator = m.group(4).upper()
+
+        if numerator == "K":
+            value = value * 1024
```

Comment:
This (and below) should be multiplying by 1000.  There's a difference between KB and KiB.  KB is 1000 and KiB is 1024.  See this article

https://www.linkedin.com/pulse/20140814132922-176099595-mb-vs-mib-gb-vs-gib/

For clarity, maybe you should write these out in using powers.  For example, this should be 10^3 and M should be 10^6, etc.